### PR TITLE
chore(build): add support for multi-arch builds

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

**What this PR does?**:
This PR adds support for multi-arch builds using github actions.

**Any additional information for your reviewer?** : 
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated? 
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track: 
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -119,7 +119,7 @@ endif
 export DBUILD_ARGS=--build-arg DBUILD_DATE=${DBUILD_DATE} --build-arg DBUILD_REPO_URL=${DBUILD_REPO_URL} --build-arg DBUILD_SITE_URL=${DBUILD_SITE_URL} --build-arg ARCH=${ARCH}
 
 # Specify the name of cstor-base image
-CSTOR_BASE_IMAGE= ${IMAGE_ORG}/cstor-base:${BASE_TAG}
+CSTOR_BASE_IMAGE_AMD64= ${IMAGE_ORG}/cstor-base-amd64:${BASE_TAG}
```
Comment (kmova) [inline GNUmakefile:122]: this change may not be required?

Comment (shubham14bajpai) [reply GNUmakefile:122]: We made this change in libcstor as well for the single arch image so made here.

Comment (kmova) [reply GNUmakefile:122]: Also any changes required to make sure travis pushes images with arch suffix.

File: .travis.yml [hunk 2]
```diff
@@ -10,6 +10,8 @@ env:
     - MINIKUBE_HOME=$HOME
     - CHANGE_MINIKUBE_NONE_USER=true
     - KUBECONFIG=$HOME/.kube/config
+    - RELEASE_TAG=$TRAVIS_TAG
+    - BRANCH=$TRAVIS_BRANCH
```
Comment (kmova) [inline .travis.yml:14]: We can also remove the arm job here.

File: GNUmakefile [hunk 3]
```diff
@@ -119,7 +119,7 @@ endif
...
 export CSTOR_BASE_IMAGE
 
 ifeq (${CSTOR_BASE_IMAGE_ARM64}, )
```
Comment (akhilerm) [inline GNUmakefile:125]: All these lines can be removed.

File: GNUmakefile [hunk 4]
```diff
@@ -119,7 +119,7 @@ endif
 export DBUILD_ARGS=--build-arg DBUILD_DATE=${DBUILD_DATE} --build-arg DBUILD_REPO_URL=${DBUILD_REPO_URL} --build-arg DBUILD_SITE_URL=${DBUILD_SITE_URL} --build-arg ARCH=${ARCH}
 
 # Specify the name of cstor-base image
-CSTOR_BASE_IMAGE= ${IMAGE_ORG}/cstor-base:${BASE_TAG}
+CSTOR_BASE_IMAGE_AMD64= ${IMAGE_ORG}/cstor-base-amd64:${BASE_TAG}
 export CSTOR_BASE_IMAGE
```
Comment (akhilerm) [inline GNUmakefile:123]: Are we using base image with amd64 suffix?, or using the {ARCH-SUFFIX}?

q: This variable will be used only by travis build process . right?

```suggestion
export CSTOR_BASE_IMAGE_AMD64
```

Comment (shubham14bajpai) [reply GNUmakefile:123]: Yes only amd64 and only travis builds will use it

File: buildscripts/admission-server/Makefile.mk [hunk 5]
```diff
@@ -3,7 +3,7 @@
 WEBHOOK=admission-server
 
 # Specify the name of the docker repo for amd64
-ADMISSION_SERVER_REPO_NAME?=admission-server
+ADMISSION_SERVER_REPO_NAME?=admission-server-amd64
 
 # Specify the name of the docker repo for arm64
 ADMISSION_SERVER_REPO_NAME_ARM64?=admission-server-arm64
```
Comment (akhilerm) [inline buildscripts/admission-server/Makefile.mk:9]: This can be removed

File: buildscripts/admission-server/admission-server.Dockerfile [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+FROM ubuntu:16.04
+
+RUN apt-get update && apt-get install -y \
+    iproute2
+
+ARG DBUILD_DATE
+ARG DBUILD_REPO_URL
+ARG DBUILD_SITE_URL
+LABEL org.label-schema.name="admission-server"
+LABEL org.label-schema.description="webhook admission server policy for OpenEBS"
```
Comment (akhilerm) [inline buildscripts/admission-server/admission-server.Dockerfile:48]:
```suggestion
LABEL org.label-schema.description="Webhook admission server policy for OpenEBS"
```

File: buildscripts/apiserver/Makefile.mk [hunk 7]
```diff
@@ -16,7 +16,7 @@
 APISERVER=maya-apiserver
 
 # Specify the name of the docker repo for amd64
-M_APISERVER_REPO_NAME?=m-apiserver
+M_APISERVER_REPO_NAME?=m-apiserver-amd64
 
 # Specify the name of the docker repo for arm64
 M_APISERVER_REPO_NAME_ARM64?=m-apiserver-arm64
```
Comment (akhilerm) [inline buildscripts/apiserver/Makefile.mk:22]: This can be removed.

File: buildscripts/apiserver/apiserver.Dockerfile [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+LABEL org.label-schema.description="API server for OpenEBS"
+LABEL org.label-schema.schema-version="1.0"
+LABEL org.label-schema.build-date=$DBUILD_DATE
+LABEL org.label-schema.vcs-url=$DBUILD_REPO_URL
+LABEL org.label-schema.url=$DBUILD_SITE_URL
+
+RUN mkdir -p /etc/apiserver/orchprovider
+RUN mkdir -p /etc/apiserver/specs
+
+COPY --from=build /go/src/github.com/openebs/maya/buildscripts/apiserver/demo-vol1.yaml /etc/apiserver/specs/
```
Comment (akhilerm) [inline buildscripts/apiserver/apiserver.Dockerfile:66]: Why is this demo file required?

Comment (kmova) [reply buildscripts/apiserver/apiserver.Dockerfile:66]: We can remove this. This was some old code which is not used anymore.

File: buildscripts/upgrade/upgrade.Dockerfile [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+RUN make upgrade
+
+FROM alpine:3.11.5
+
+ARG DBUILD_DATE
+ARG DBUILD_REPO_URL
+ARG DBUILD_SITE_URL
+LABEL org.label-schema.name="upgrade"
+LABEL org.label-schema.description="upgrades openebs components"
```
Comment (akhilerm) [inline buildscripts/upgrade/upgrade.Dockerfile:46]:
```suggestion
LABEL org.label-schema.description="Upgrades openebs components"
```

**Review by akhilerm — COMMENTED** [review]
@shubham14bajpai Given some comments,

Also in all the component specific makefiles , the arm related code can be removed.

**Comment by shubham14bajpai** [comment]
> @shubham14bajpai Given some comments,
> 
> Also in all the component specific makefiles , the arm related code can be removed.

Addressed a few comments. Will clean up the different arch build related code once the multi-arch images pass the tests in separate PR.

**Review by akhilerm — APPROVED** [review]
/lgtm

The arch related code will be cleaned up in a separate PR.

**Review by kmova — APPROVED** [review]

**Pull request merged**